ROX-12341: Add support for creating resolver instance with context

keyallis opened the pull request:
## Description

Pass resolver context to sub resolver for usage and ensure sub resolvers use parent context.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added

## Testing Performed

Manually verified some graphql endpoints to ensure nested objects resolve as anticipated.

md2119 reviewed (commented):
Update the following conditionally for Postgres.
```
func (resolver *clusterResolver) NodeVulnerabilities(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error)

func (resolver *clusterResolver) NodeVulnerabilityCount(ctx context.Context, args RawQuery) (int32, error) 

func (resolver *clusterResolver) NodeVulnerabilityCounter(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error)
```

keyallis commented:
Did a rebase on master since changes were going to be made anyways. Found a regression issue in non-postgress causing a panic and addressed. Addressed above comment.

keyallis commented on central/graphql/resolvers/cluster_vulnerabilities.go:
```diff
@@ -72,11 +73,10 @@ func (resolver *Resolver) ClusterVulnerability(ctx context.Context, args IDQuery
 
 	ret, err := loader.FromID(ctx, string(*args.ID))
-	vulnResolver, err := resolver.wrapClusterCVE(ret, true, err)
+	vulnResolver, err := resolver.wrapClusterCVEWithContext(ctx, ret, true, err)
```
Good catch

keyallis commented on central/graphql/resolvers/cluster_vulnerabilities.go:
```diff
@@ -316,14 +316,14 @@ func withOpenShiftTypeFiltering(q string) string {
 		})
 	}
-	return scoped.Context(ctx, scoped.Scope{
+	return scoped.Context(resolver.ctx, scoped.Scope{
```
Given that the UI changes and the BE changes are dependent on Postgres, there is no longer any requirement to support new endpoints without Postgres other than for the sake of convenience to front end development (which is not necessary since the BE changes now work with Postgres enabled)

keyallis commented:
/retest

md2119 reviewed (approved):
Apart from concerns about cluster scope (potentially namespace scope), changes LGTM.
Test the following flows for err and correctness:
- cluster -> nodes
- cluster -> deployment
- cluster -> ns -> deployment
- cluster -> image -> vulns
- cluster -> node -> vulns

There is a known issue with cluster scope context for non-postgres setting hat Charmik pointed out. Therefore, I recommend feature flag it. Otherwise, this PR must be reverted before 3.73 release if Postgres is not shipped in 3.73, which honestly is risky.
